src/overlays/actors/ovl_En_Zl2/z_en_zl2.c

```diff
@@ -27,265 +101,1664 @@ const ActorInit En_Zl2_InitVars = {
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+}
+
+void func_80B4E9B0(EnZl2* this) {
+    s32 pad[4];
+    s16* unk_194 = &this->unk_194;
+    s16* unk_192 = &this->unk_192;
+    s16* unk_190 = &this->unk_190;
+
+    if (DECR(*unk_192) == 0) {
+        *unk_192 = Math_Rand_S16Offset(0x3C, 0x3C);
+    }
+    *unk_190 = *unk_192;
+    if (*unk_190 >= 3) {
+        *unk_190 = 0;
+    }
+    *unk_194 = *unk_190;
+}
+
+void func_80B4EA40(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 3.0f) {
+        *unk_190 = 0;
+    } else {
+        if (*unk_27C < 6.0f) {
+            *unk_190 = 1;
+        } else if (*unk_27C < 9.0f) {
+            *unk_190 = 2;
+        } else {
+            *unk_190 = 5;
+            this->unk_194 = *unk_190;
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4EAF4(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 2.0f) {
+        *unk_190 = 5;
+    } else {
+        if (*unk_27C < 4.0f) {
+            *unk_190 = 2;
+        } else if (*unk_27C < 6.0f) {
+            *unk_190 = 1;
+        } else {
+            func_80B4E9B0(this);
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4EBB8(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 2.0f) {
+        *unk_190 = 0;
+    } else {
+        if (*unk_27C < 4.0f) {
+            *unk_190 = 1;
+        } else {
+            *unk_190 = 2;
+            this->unk_194 = *unk_190;
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4EC48(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 2.0f) {
+        *unk_190 = 2;
+    } else {
+        if (*unk_27C < 4.0f) {
+            *unk_190 = 1;
+        } else {
+            func_80B4E9B0(this);
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4ECE8(EnZl2* this, s16 arg1) {
+    this->unk_190 = arg1;
+    this->unk_194 = this->unk_190;
+}
+
+void func_80B4ED04(EnZl2* this, s16 arg1) {
+    this->unk_194 = arg1;
+}
+
+void func_80B4ED18(EnZl2* this, s16 arg1) {
+    this->unk_198 = arg1;
+}
+
+void func_80B4ED2C(EnZl2* this, GlobalContext* globalCtx) {
+    func_8002E4B4(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 5);
+}
+
+s32 EnZl2_FrameUpdateMatrix(EnZl2* this) {
+    return SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+CsCmdActorAction* func_80B4ED94(GlobalContext* globalCtx, s32 idx) {
+    if (globalCtx->csCtx.state != 0) {
+        return globalCtx->csCtx.npcActions[idx];
+    }
+    return NULL;
+}
+
+void func_80B4EDB8(EnZl2* this, GlobalContext* globalCtx, s32 arg2) {
+    CsCmdActorAction* npcAction = func_80B4ED94(globalCtx, arg2);
```

Comment:
probably _SetPosRot, dont know if you want to fit the fact that its the position and rotation for the cutscene into the name. its probably ok without it

Reply:
func_80B50278 does a similar thing but with a specific action index. I'm not sure on naming either of these